### PR TITLE
MINOR: Fix producer Callback comment

### PR DESCRIPTION
**Pull request by fvaleri**
Fixes the wrong exception name: OffsetMetadataTooLarge~~Exception~~.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/producer/Callback.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public interface Callback {
      *                  Non-Retriable exceptions (fatal, the message will never be sent):
      *                  <ul>
      *                  <li>InvalidTopicException
-     *                  <li>OffsetMetadataTooLargeException
+     *                  <li>OffsetMetadataTooLarge
```
Comment (machi1990) [inline clients/src/main/java/org/apache/kafka/clients/producer/Callback.java]: good catch!

A more general comment unrelated to the change, do you think we can link the exceptions classes?

Comment (fvaleri) [reply clients/src/main/java/org/apache/kafka/clients/producer/Callback.java]: Hi @machi1990, that's good idea. 

I think what you propose would help in preventing errors like this. Done.

**Review by machi1990 — APPROVED** [review]
lgtm

good catch, thanks @fvaleri

**Review by showuon — APPROVED** [review]

**Comment by showuon** [comment]
CI build failed with infra issue. Rebuilding
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13669/3/

**Pull request merged**